markets/bfp-market/test/integration/modules/OrderModule.fees.test.ts

```diff
@@ -0,0 +1,310 @@
+        const takerFeeUsd = wei(order1.sizeDelta.abs()).mul(order2.fillPrice).mul(takerFee);
+
+        assertBn.equal(orderFee2, makerFeeUsd.add(takerFeeUsd).toBN());
+      });
+
+      it('should revert when marketId does not exist', async () => {
+        const { PerpMarketProxy } = systems();
+
+        const invalidMarketId = 42069;
+        await assertRevert(
+          PerpMarketProxy.getOrderFees(invalidMarketId, bn(0), bn(0)),
+          `MarketNotFound("${invalidMarketId}")`,
+          PerpMarketProxy
+        );
+      });
+    });
+
+    describe('keeperFee', () => {
+      const getKeeperFee = (
+        PerpMarketProxy: PerpMarketProxy,
+        receipt: ethers.ContractReceipt
+      ): BigNumber => findEventSafe(receipt, 'OrderSettled', PerpMarketProxy)?.args.keeperFee;
+
+      it('should calculate keeper fees proportional to block.baseFee and profit margin', async () => {
+        const { PerpMarketProxy } = systems();
+
+        const { trader, marketId, market, collateral, collateralDepositAmount } =
+          await depositMargin(bs, genTrader(bs));
+        const order = await genOrder(bs, market, collateral, collateralDepositAmount);
+        const { orderFee } = await PerpMarketProxy.getOrderFees(
+          marketId,
+          order.sizeDelta,
+          order.keeperFeeBufferUsd
+        );
+        const { calcKeeperOrderSettlementFee } = await calcOrderFees(
+          bs,
+          marketId,
+          order.sizeDelta,
+          order.keeperFeeBufferUsd
+        );
+        const { tx, receipt, lastBaseFeePerGas } = await commitAndSettle(
+          bs,
+          marketId,
+          trader,
+          order
+        );
+
+        const keeperFee = getKeeperFee(PerpMarketProxy, receipt);
+        const expectedKeeperFee = calcKeeperOrderSettlementFee(lastBaseFeePerGas);
+        assertBn.equal(expectedKeeperFee, keeperFee);
+        const block = await provider().getBlock(receipt.blockNumber);
+        const timestamp = block.timestamp;
+
+        await assertEvent(
+          tx,
+          `OrderSettled(${trader.accountId}, ${marketId}, ${timestamp}, ${order.sizeDelta}, ${orderFee}, ${expectedKeeperFee}, 0, 0, 0, ${order.fillPrice}, 0)`,
+          PerpMarketProxy
+        );
+      });
+
+      it('should cap the keeperFee by its max usd when exceeds ceiling', async () => {
+        const { PerpMarketProxy } = systems();
+
+        // Cap the max keeperFee to $50 USD
+        const maxKeeperFeeUsd = bn(50);
+        await setMarketConfiguration(bs, { maxKeeperFeeUsd, minKeeperFeeUsd: bn(10) });
+
+        const { trader, marketId, market, collateral, collateralDepositAmount } =
+          await depositMargin(bs, genTrader(bs));
+        const order = await genOrder(bs, market, collateral, collateralDepositAmount, {
+          desiredKeeperFeeBufferUsd: 1000, // Set a really high keeper buffer fee so we hit the max amount.
```

Comment:
Would like to see a test where the keeperFee paid wasn't inflated by a large `desiredKeeperFeeBufferUsd` but rather due to expensive gas.

Reply:
Yea I was running into weird anvil edgecases with gas, this felt more stable. But maybe its more stable with the hardcoded gaslimits for settle and commit.

Reply:
Fixed